builder/builder.go

```diff
@@ -639,6 +639,157 @@ func (b *Builder) BuildUpdate(prefixflag string, minVersion int, format string,
 	return nil
 }
 
+func (b *Builder) buildUpdateWithNewSwupd(timer *stopWatch, mixVersion uint32, minVersion uint32, format uint32, skipSigning bool) error {
+	var err error
+
+	outputDir := filepath.Join(b.Statedir, "www")
+	if _, err = os.Stat(outputDir); err != nil {
+		return errors.Wrapf(err, "couldn't access the swupd output dir (www)")
+	}
+
+	timer.Start("CREATE MANIFESTS")
+	mom, err := swupd.CreateManifests(mixVersion, minVersion, uint(format), b.Statedir)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create update metadata")
+	}
+	fmt.Printf("MoM version %d\n", mom.Header.Version)
+	for _, f := range mom.Files {
+		fmt.Printf("- %-20s %d\n", f.Name, f.Version)
+	}
+
+	if skipSigning {
+		fmt.Println("SKIPPING manifest signing.")
+	} else {
+		fmt.Println("Signing manifest.")
+		err = b.SignManifestMoM()
+		if err != nil {
+			return err
+		}
+	}
+
+	thisVersionDir := filepath.Join(outputDir, fmt.Sprint(mixVersion))
+	fmt.Println("Compressing Manifest.MoM")
+	momF := filepath.Join(thisVersionDir, "Manifest.MoM")
+	if skipSigning {
+		err = createCompressedArchive(momF+".tar", momF)
+	} else {
+		err = createCompressedArchive(momF+".tar", momF, momF+".sig")
+	}
+	if err != nil {
+		return err
+	}
+	fmt.Println("Compressing bundle manifests")
+	for _, bundle := range mom.UpdatedBundles {
+		fmt.Printf("  %s\n", bundle.Name)
+		f := filepath.Join(thisVersionDir, "Manifest."+bundle.Name)
+		err = createCompressedArchive(f+".tar", f)
+		if err != nil {
+			return err
+		}
+	}
+	// TODO: Create manifest tars for Manifest.MoM and the mom.UpdatedBundles.
+	timer.Stop()
+
+	timer.Start("CREATE FULLFILES")
+	fullfilesDir := filepath.Join(outputDir, b.Mixver, "files")
+	fullChrootDir := filepath.Join(b.Statedir, "image", b.Mixver, "full")
+	// TODO: CreateFullfiles should return us feedback on what was
+	// done so we can report here.
+	err = swupd.CreateFullfiles(mom.FullManifest, fullChrootDir, fullfilesDir)
+	if err != nil {
+		return err
+	}
+	timer.Stop()
+
+	timer.Start("CREATE ZERO PACKS")
+	for _, bundle := range mom.Files {
```

Comment:
Is this worth being a function, maybe `CreatePacks` or `CreateAllPacks`, in the swupd library?

Reply:
We do take advantage of printing output during the loop in this code, so moving to the library we will have to decide how to handle that (e.g. pass a io.Writer, etc). I rather do this when we explore using goroutines, as one might influence another (having a channel to get the intermediate information).